delphin/util.py

```diff
@@ -272,3 +272,43 @@ def peek(self, n=0, skip=None, drop=False):
             self._buffer_fill(n + 1)
             datum = buffer[n]
         return datum
+
+
+# modified from https://www.python.org/dev/peps/pep-0263/#defining-the-encoding
+_encoding_symbol_re = re.compile(
+    r'^[ \t\f]*.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', re.IGNORECASE)
```

Comment:
I see you've dropped the comment character from the regex. This is fine (since it's configurable), but see my comment below.
Alternatively, you could do `re.match()` within the function and put the specified comment character in the pattern itself.